File: libs/monobitmap-manager/src/test/kotlin/mono/bitmap/manager/factory/RectangleBitmapFactoryTest.kt
package mono.bitmap.manager.factory

import mono.graphics.geo.Rect
import mono.shape.shape.Rectangle
import kotlin.test.Test
import kotlin.test.assertEquals

/**
 * A test for [RectangleBitmapFactory]
 */
class RectangleBitmapFactoryTest {
    @Test
    fun testToBitmap() {
        val shape = Rectangle(Rect.byLTWH(10, 10, 5, 5))
        val bitmap = RectangleBitmapFactory.toBitmap(shape.bound.size, shape.extra)
        assertEquals(shape.bound.width, bitmap.size.width)
        assertEquals(shape.bound.height, bitmap.size.height)
        assertEquals(
            """
            ┌───┐
            │   │
            │   │
            │   │
            └───┘
            """.trimIndent(),
            bitmap.toString()
        )
    }
}

File: monosketch-svelte/src/lib/mono/monobitmap/manager/rectangle-bitmap-factory.test.ts
/*
 * Copyright (c) 2024, tuanchauict
 */

import { Rect } from "$libs/graphics-geo/rect";
import { StringExt } from "$libs/string";
import { describe, expect, test } from "vitest";
import { RectangleBitmapFactory } from './rectangle-bitmap-factory';
import { Rectangle } from '$mono/shape/shape/rectangle';
import trimIndent = StringExt.trimIndent;

describe('RectangleBitmapFactory', () => {
    test('toBitmap', () => {
        const shape = Rectangle.fromRect({ rect: Rect.byLTWH(10, 10, 5, 5) });
        const bitmap = RectangleBitmapFactory.toBitmap(shape.bound.size, shape.extra);
        expect(bitmap.size.width).toBe(shape.bound.width);
        expect(bitmap.size.height).toBe(shape.bound.height);
        console.log(bitmap.toString());
        expect(bitmap.toString().trim()).toBe(
            trimIndent(`
            ┌───┐
            │   │
            │   │
            │   │
            └───┘
            `),
        );
    });
});
